src/pages/create-guild.tsx

```diff
@@ -1,98 +1,28 @@
-  } = useCreateGuildContext()
-  const { control } = useFormContext<GuildFormType>()
-
-  const name = useWatch({ name: "name" })
-  const imageUrl = useWatch({ name: "imageUrl" })
-  const contacts = useWatch({ name: "contacts" })
-  const { captureEvent } = usePostHogContext()
-
-  const themeColor = useWatch({ name: "theme.color" })
-  const color = localThemeColor !== themeColor ? themeColor : localThemeColor
-  const isLastSubStep = STEPS[activeStep].progress.length === stepPart + 1
-
-  const nextWithPostHog = () => {
-    // +2 because, index starts with 0 and we jump to the step after the active one
-    captureEvent("guild creation flow > continue", { to: activeStep + 2 })
-
-    // email can be added in step 2
-    if (activeStep === 1 && contacts[0].contact)
-      captureEvent("guild creation flow > contacts added")
-
-    nextStep()
-  }
-
-  useEffect(() => {
-    captureEvent("guild creation flow > start guild creation")
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
-
-  return (
-    <>
-      <Layout
-        title={name || "Create Guild"}
-        backgroundOffset={47}
-        textColor={textColor}
-        background={color}
-        backgroundImage={localBackgroundImage}
-        image={
-          imageUrl && (
-            <GuildLogo
-              imageUrl={imageUrl}
-              size={{ base: "56px", lg: "72px" }}
-              mt={{ base: 1, lg: 2 }}
-              bgColor={textColor === "primary.800" ? "primary.800" : "transparent"}
-            />
-          )
-        }
-        imageUrl={imageUrl}
-        showFooter={false}
-      >
-        <CreateGuildStepper
-          {...{ color, activeStep, setActiveStep, textColor, stepPart }}
-        />
-        <Stack w="full" spacing={4} pt={STEPS[activeStep].content ? 6 : 0} pb="24">
-          {STEPS[activeStep].content}
-        </Stack>
-        <GuildCreationProgress
-          next={isLastSubStep ? nextWithPostHog : () => setPart(stepPart + 1)}
-          progress={STEPS[activeStep].progress[stepPart]}
-          isDisabled={nextStepIsDisabled}
-        >
-          {stepPart === 1 ? <CreateGuildButton /> : null}
-        </GuildCreationProgress>
-      </Layout>
-      <DynamicDevTool control={control} />
-    </>
-  )
-}
+import { ThemeProvider } from "components/[guild]/ThemeContext"
+import { Layout } from "components/common/Layout"
+import { CreateGuildBackground } from "components/create-guild/CreateGuildBackground"
+import { CreateGuildContent } from "components/create-guild/CreateGuildContent"
+import { CreateGuildProvider } from "components/create-guild/CreateGuildContext"
+import { CreateGuildDynamicDevTool } from "components/create-guild/CreateGuildDynamicDevTool"
+import { CreateGuildHead } from "components/create-guild/CreateGuildHead"
+import { CreateGuildHeadline } from "components/create-guild/CreateGuildHeadline"
+
+const CreateGuildPage = (): JSX.Element => (
+  <>
+    <Layout.Root>
+      <CreateGuildHead />
```

Comment:
It's really great that the CreateGuildContent and the general layout logic is separated now, but putting each layout component in a separate wrapper component is unnecessary, I find it too much abstraction. We should simply just inline them here, so the page structure (including related logic and props) is easy to see through, just keep CreateGuildContent separately. If the Head and Headline components need to be separate too because of the performance problems, keep them as one separate component too (we can put Head in Headline as NextHead can be rendered anywhere)

Reply:
> just keep CreateGuildContent separately

Do you mean the other `CreateGuild` components should reside in the `create-guild.tsx`? 

We need to have separate wrapper components for them because they could update each other unnecessarily.